### DISCUSSION TITLE
What makes an inference microservice ill-defined?

### DISCUSSION THREAD

**Discussion by kapilagrawal95** [facebookresearch/distributed_traces#3] [Q&A] [answered]
Hi,

Thanks for the paper! In section 3, you categorize meta's microservices into three categories: i. well-defined microservice for a single business use-case, ii. multiple business use-cases, iii. ill-defined microservices. Later in section 3.1, you expand on two examples of ill-defined microservices: i. Inference platform, ii. ML scheduler. I am trying to understand what makes an inference service ill-defined (in the context of their business use-case)? For example, in 4.1, you provide details on the RaaS service. Is RaaS an example of an ill-defined microservice? If yes, why? If no, is it possible to give one example so I can clarify my understanding a little bit more?

**Comment by yurishkuro** [comment, marked as answer]
These two platforms are examples of two different classes of ill-fitting microservices:
  1. Inference platform _explodes_ the Service ID space by assigning unique Service ID to each model, which are generally short-lived. My guess is many of these model IDs actually refer the same logical use case, just different versions, but the way the platform chose to represent that in Service ID is not reflective of that, so the "stable use case" property of microservices is not preserved.
  2. ML Scheduler _collapses_ the Service ID space into one ID, the opposite behavior of Inference platform. So a single Service ID ends up representing different distinct use cases, and the platform itself differentiates them by some other dimensions that are not reflected in the Service ID space.

**Comment by kapilagrawal95** [comment]
I see, thanks! That clears it up. A follow-up question is in 3.1, you also mention about storage systems and they take up a lot of service instances (not service IDs). Are they also among the ill-fitting microservices?

**Comment by yurishkuro** [comment]
Large number of instances does not mean a service is ill-fitting, it just relates to its scale.

**Comment by kapilagrawal95** [comment]
Thanks! Closing this.
